app/controllers/merchant/discounts_controller.rb

```diff
@@ -0,0 +1,52 @@
+class Merchant::DiscountsController < Merchant::BaseController
```

Comment:
Added a namespaced controller for the discounts which is under the control of a merchant/merchant_user.